polyfill/test/duration.mjs

```diff
@@ -268,16 +341,29 @@ describe('Duration', () => {
       equal(result.microseconds, 3);
       equal(result.nanoseconds, 1);
     });
-    it('negative values always throw', () => {
+    it('mixed positive and negative values always throw', () => {
       ['constrain', 'balance', 'reject'].forEach((disambiguation) =>
-        throws(() => duration.with({ minutes: -1 }, { disambiguation }), RangeError)
+        throws(() => duration.with({ hours: 1, minutes: -1 }, { disambiguation }), RangeError)
       );
     });
+    it('can reverse the sign if all the fields are replaced', () => {
+      const d = Duration.from({ years: 5, days: 1 });
+      const d2 = d.with({ years: -1, days: -1, minutes: 0 });
+      equal(`${d2}`, '-P1Y1D');
+      notEqual(d.sign, d2.sign);
+    });
+    it('throws if new fields have a different sign from the old fields', () => {
+      const d = Duration.from({ years: 5, days: 1 });
+      throws(() => d.with({ months: -5, minutes: 0 }), RangeError);
+    });
     it('invalid disambiguation', () => {
       ['', 'CONSTRAIN', 'xyz', 3, null].forEach((disambiguation) =>
         throws(() => duration.with({ day: 5 }, { disambiguation }), RangeError)
       );
     });
+    it('sign cannot be manipulated independently', () => {
+      throws(() => duration.with({ sign: -1 }), RangeError);
```

Comment:
I thought `Duration.prototype.with` ignores unknown properties, per #325.  I think it's a good idea to throw here to prevent bugs, but what about (singular) 'year', 'month', etc ?  Is `'sign'` special-cased?

Reply:
What's happening here is that it will ignore unknown properties if they are in addition to known properties, but if you pass an empty object or an object with only unknown properties, then it will throw.

Reply:
OK. Is this new behavior with this PR? Or simply leveraging existing behavior of `with`?

BTW, isn't the test case above this one (line 278) wrong because `.with({ day: 5 }...` will always throw regardless of `disambiguation` value?  Shouldn't it be `.with({ days: 5 }...`?

Reply:
That's the existing behaviour of `with`.

Good catch! Maybe I should split the "various fixes" out into a different PR, and add that one as well.

Reply:
#815